### PR TITLE
Add new APIs to read EC public and private keys from files

### PR DESCRIPTION
**Pull request by Bhashinee**
## Purpose
Fixes: https://github.com/ballerina-platform/ballerina-library/issues/5871

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests
- [x] Updated the spec
- [ ] Checked native-image compatibility

### CODE REVIEW

File: ballerina/tests/private_public_key_test.bal [hunk 1]
```diff
@@ -146,6 +146,32 @@ isolated function testParsePrivateKeyFromKeyPairFile() returns Error? {
     test:assertEquals(result["algorithm"], "RSA");
 }
 
+@test:Config {}
+isolated function testParseEcPrivateKeyFromKeyFile() returns Error? {
+    PrivateKey result = check decodeEcPrivateKeyFromKeyFile(EC_PRIVATE_KEY_PATH);
+    test:assertEquals(result["algorithm"], "ECDSA");
```
Comment (MohamedSabthar) [inline ballerina/tests/private_public_key_test.bal:152]: can't we access this with dot operator `result.algorithm` ?

File: docs/spec/spec.md [hunk 2]
```diff
@@ -32,11 +32,15 @@ The conforming implementation of the specification is released and included in t
...
+   * 4.1. [Decode RSA Private key from PKCS12 file](#41-rsa-decode-private-key-from-pkcs12-file)
    * 4.2. [Decode RSA Private key using Private key and Password](#42-decode-rsa-private-key-using-private-key-and-password)
    * 4.3. [Decode RSA Public key from PKCS12 file](#43-decode-rsa-public-key-from-pkcs12-file)
    * 4.4. [Decode RSA Public key from the certificate file](#44-decode-rsa-public-key-from-the-certificate-file)
-   * 4.5. [Build RSA Public key from modulus and exponent parameters](#45-build-rsa-public-key-from-modulus-and-exponent-parameters)
+   * 4.5. [Decode EC Private key from PKCS12 file](#45-decode-ec-private-key-from-pkcs12-file)
+   * 4.6. [Decode EC Private key using Private key and Password](#46-decode-ec-private-key-using-private-key-and-password)
+   * 4.7. [Decode EC Public key from PKCS12 file](#47-decode-ec-public-key-from-pkcs12-file)
+   * 4.8. [Decode EC Public key from the certificate file](#48-decode-ec-public-key-from-the-certificate-file)
+   * 4.9. [Build RSA Public key from modulus and exponent parameters](#49-build-rsa-public-key-from-modulus-and-exponent-parameters)
```
Comment (MohamedSabthar) [inline docs/spec/spec.md:43]: shall we keep RSA topics together?  can't we move this after 4.4 ?

File: native/src/main/java/io/ballerina/stdlib/crypto/nativeimpl/Decode.java [hunk 3]
```diff
@@ -151,7 +167,7 @@ public static Object decodeRsaPrivateKeyFromKeyFile(BString keyFilePath, Object
                         keyFilePath.getValue());
             }
             PrivateKey privateKey = converter.getPrivateKey(privateKeyInfo);
-            return buildRsPrivateKeyRecord(privateKey);
+            return privateKey;
```
Comment (MohamedSabthar) [inline native/src/main/java/io/ballerina/stdlib/crypto/nativeimpl/Decode.java:170]: return converter.getPrivateKey(privateKeyInfo);

File: docs/spec/spec.md (hunk 2 above)
Comment (Bhashinee) [reply docs/spec/spec.md:43]: Thought of keeping decoding APIs together. Hence, kept the `build` API at last.

**Review by MohamedSabthar — APPROVED** [review]

**Pull request merged**